openpype/modules/kitsu/plugins/publish/integrate_kitsu_note.py

```diff
@@ -21,30 +21,32 @@ def process(self, context):
 
         self.log.debug("Comment is `{}`".format(publish_comment))
 
-        # Get note status, by default uses the task status for the note
-        # if it is not specified in the configuration
-        note_status = context.data["kitsu_task"]["task_status_id"]
-        if self.set_status_note:
-            kitsu_status = gazu.task.get_task_status_by_short_name(
-                self.note_status_shortname
-            )
-            if kitsu_status:
-                note_status = kitsu_status
-                self.log.info("Note Kitsu status: {}".format(note_status))
-            else:
-                self.log.info(
-                    "Cannot find {} status. The status will not be "
-                    "changed!".format(self.note_status_shortname)
-                )
+        for instance in context:
+
+            # Get note status, by default uses the task status for the note
+            # if it is not specified in the configuration
+            note_status = instance.data["kitsu_task"]["task_status"]["id"]
 
-        # Add comment to kitsu task
-        self.log.debug(
-            "Add new note in taks id {}".format(
-                context.data["kitsu_task"]["id"]
+            if self.set_status_note:
+                kitsu_status = gazu.task.get_task_status_by_short_name(
+                    self.note_status_shortname
+                )
+                if kitsu_status:
+                    note_status = kitsu_status
+                    self.log.info("Note Kitsu status: {}".format(note_status))
+                else:
+                    self.log.info(
+                        "Cannot find {} status. The status will not be "
+                        "changed!".format(self.note_status_shortname)
+                    )
+
+            # Add comment to kitsu task
+            task = instance.data["kitsu_task"]["id"]
+            self.log.debug(
+                "Add new note in taks id {}".format(task)
+            )
+            kitsu_comment = gazu.task.add_comment(
+                task, note_status, comment=publish_comment
```

Comment:
You've changed what is passed in. It was passing object of task and not you're passing only it's id. Is it Ok?

Reply:
Yes, add_comment() ether expects a dict or a ID and with the new code it was easier to just pass the ID